fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -216,10 +216,10 @@ public long beginTransaction(long dbId, String label, TUniqueId requestId,
     private void checkRunningTxnExceedLimit(long dbId, LoadJobSourceType sourceType) throws BeginTransactionException {
         switch (sourceType) {
             case ROUTINE_LOAD_TASK:
-                // we do not limit the txn num of routine load here. for 2 reasons:
-                // 1. the number of running routine load tasks is limited by Config.max_routine_load_task_num_per_be
-                // 2. if we add routine load txn to runningTxnNums, runningTxnNums will always be occupied by routine load,
-                //    and other txn may not be able to submitted.
+                if (runningRoutineLoadTxnNums.getOrDefault(dbId, 0) >= Config.max_running_txn_num_per_db) {
```

Comment:
The task will not be renew before the txn is visible at above. So we do not limit the txn in here

Reply:
I prefer to add the double check here, in case we meet some other unexpected bugs.

Reply:
Maybe you are right, if I add this check here, may hurt some current routine load job user.
I will remove it.